Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1931,3 +1931,29 @@
     UNLOCK_CAIRO();
     return returnValue;
 }
+CGPathRef CGContextCairo::CGContextCopyPath(void) {
+    CGMutablePathRef copyPath = CGPathCreateMutable();
+    cairo_path_t* caPath = cairo_copy_path(_drawContext);
```

Comment:
>cairo_copy_path [](start = 27, length = 15)

Does this have to be done under CAIRO_LOCK()? presumably path could be changed while we are attempting to copy?